ee/katc/config.go

```diff
@@ -16,8 +15,11 @@ import (
 // identifier parsed from the JSON KATC config, and the `dataFunc` is the function
 // that performs the query against the source.
 type katcSourceType struct {
-	name     string
-	dataFunc func(ctx context.Context, slogger *slog.Logger, sourcePattern string, query string, sourceConstraints *table.ConstraintList) ([]sourceData, error)
+	name string
+	// `pathConstraints` comes from the live query or scheduled query, and constrains `sourcePaths` to a particular value.
```

Comment:
I'm still confused about what `pathConstraints` achieve that globs in `sourcePaths` do not. 

I might offer that if `sourcePaths` is a list of possible globs, the only thing it _cannot_ do is have negative paths. Like if you had `/Users/%/path/to/file` and wanted to omit `root`, or `/Users/%/.config/netrc%` and wanted to omit `netrc~`. So something as a negative glob can make sense. But I don't understand as a further narrowing. 

Also don't know that the juice is worth the squeeze yet.

Reply:
Becca got me in a huddle and said this is `queryConstraints`, from the query -- it's an optimization. We discussed some renaming to make that clear.